Sources/ApolloCodegenLib/Templates/MockObjectTemplate.swift

```diff
@@ -63,14 +63,31 @@ struct MockObjectTemplate: TemplateRenderer {
             """ }, separator: ",\n")
         ) {
           self.init()
-          \(fields.map { "_set(\($0.initializerParameterName ?? $0.propertyName), for: \\.\($0.propertyName))" }, separator: "\n")
+          \(fields.map {
+            return "_set\(mockFunctionDescriptor($0.type))(\($0.initializerParameterName ?? $0.propertyName), for: \\.\($0.propertyName))"
+          }, separator: "\n")
         }
       }
       """) : TemplateString(stringLiteral: "")
     )
     
     """
   }
+  
+  private func mockFunctionDescriptor(_ graphQLType: GraphQLType) -> String {
+    switch graphQLType {
+      case .list(_):
+        return "List"
+      case .scalar(_), .enum(_):
+        return "Scalar"
+      case .entity(_):
+        return "Entity"
+      case .inputObject(_):
+        preconditionFailure()
```

Comment:
Nit: might want to also add a message to this failure to help us identify the issue if a user ever encounters this and files a bug report.